classes/extNodeProxyDocument.sc

```diff
@@ -0,0 +1,18 @@
++ NodeProxy {
+	accessStr { | envir |
+		var accessStr = "a", nameStr, isAnon;
+
+		envir = envir ? currentEnvironment;
+
+		nameStr = envir.use { this.asCompileString };
+		isAnon = nameStr.beginsWith("a = ");
+		if (isAnon.not) { accessStr = nameStr };
+
+		^accessStr;
+	}
+
+	currentSettingsAsCompileString {
```

Comment:
settingsAsCompileString would be better. Settings are always current, just like any object.

Reply:
please try :)

Reply:
will do! 
Until this is finalized, you could do:
```
Ndef(\x, { LFSaw.ar(\freq.kr) }).set(\freq, 123);
Ndef(\x).nodeMap.asCode(Ndef(\x), true); 
```

Reply:
hello, so wow; I was not aware of `ProxyNodeMap:asCode`, somehow... 

so would

```
+ NodeProxy {
	settingsAsCode{|namestring, dropOut = true| 
		namestring = namestring ?? {this.asCompileString};
		^this.nodeMap.asCode(namestring, dropOut); 
}
```

do the trick?